subcommand/webhook-cert-manager/command.go

```diff
@@ -0,0 +1,343 @@
+		return 1
+	}
+
+	ctx := context.Background()
+	ctx, cancelFunc := context.WithCancel(ctx)
+	defer cancelFunc()
+
+	for _, config := range configs {
+		if err := config.validate(ctx, c.clientset); err != nil {
+			c.UI.Error(fmt.Sprintf("Error parsing config: %s", err))
+			return 1
+		}
+	}
+
+	certCh := make(chan cert.MetaBundle)
+
+	// Create the certificate notifier so we can update certificates,
+	// then start all the background routines for updating certificates.
+	var notifiers []*cert.Notify
+	var expiry time.Duration
+	if c.certExpiry != nil {
+		expiry = *c.certExpiry
+	} else {
+		expiry = 24 * time.Hour
+	}
+
+	for _, config := range configs {
+		certSource := &cert.GenSource{
+			Name:   "Consul Webhook Certificates",
+			Hosts:  config.TLSAutoHosts,
+			Expiry: expiry,
+		}
+		certNotify := &cert.Notify{Source: certSource, Ch: certCh, WebhookConfigName: config.Name, SecretName: config.SecretName, SecretNamespace: config.SecretNamespace}
+		notifiers = append(notifiers, certNotify)
+		go certNotify.Start(ctx)
+	}
+
+	go c.certWatcher(ctx, certCh, c.clientset, c.logger)
+
+	closeNotifiers := func() {
+		for _, notifier := range notifiers {
+			notifier.Stop()
+		}
+	}
+
+	// We define a signal handler for OS interrupts, and when an SIGINT is received,
+	// we gracefully shut down, by first stopping our cert notifiers and then cancelling
+	// all the contexts that have been created by the process.
+	select {
+	case <-c.sigCh:
+		closeNotifiers()
+		cancelFunc()
+		return 0
+	}
+}
+
+func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.MetaBundle, clientset kubernetes.Interface, log hclog.Logger) {
+	var bundle cert.MetaBundle
+	for {
+		select {
+		case bundle = <-ch:
+			log.Info(fmt.Sprintf("updated certificate bundle received for %s; updating webhook certs.", bundle.WebhookConfigName))
+			// Bundle is updated, set it up
+
+		case <-time.After(30 * time.Minute):
+			// This forces the mutating ctrlWebhook config to remain updated
+			// fairly quickly. This is done every 30 minutes to ensure the certificates
+			// are in sync. Because the certificate and key are being read from a secret,
+			// this does not have to be processed as aggressively as the 1 sec time in
+			// the connect inject cert watcher.
+
+		case <-ctx.Done():
+			// Quit
+			return
+		}
+
+		log.Info("getting secret from kubernetes")
+		certSecret, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Get(ctx, bundle.SecretName, metav1.GetOptions{})
+		if err != nil && k8serrors.IsNotFound(err) {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      bundle.SecretName,
+					Namespace: bundle.SecretNamespace,
+				},
+				Data: map[string][]byte{
+					corev1.TLSCertKey:       bundle.Cert,
+					corev1.TLSPrivateKeyKey: bundle.Key,
+				},
+				Type: corev1.SecretTypeTLS,
+			}
+
+			log.Info(fmt.Sprintf("creating Kubernetes secret with certificate for the Mutating Webhook Config %s", bundle.WebhookConfigName))
```

Comment:
```suggestion
			iterLog.Info("creating Kubernetes secret with certificate")
```
Then these can just be